eradiate/scenes/measure.py

```diff
@@ -479,33 +476,30 @@ def kernel_dict(self, ref=True):
         from eradiate.kernel.core import ScalarTransform4f
 
         target = self.target.to(kdu.get("length")).magnitude
-        distance = self.distance.to(kdu.get("length")).magnitude
-        zenith = self.zenith.to("rad").magnitude
-        azimuth = self.azimuth.to("rad").magnitude
-
-        origin = spherical_to_cartesian(distance, zenith, azimuth)
-        direction = origin / np.linalg.norm(origin)
+        origin = self.origin.to(kdu.get("length")).magnitude
 
         if np.allclose(origin, target):
             raise ValueError("target is too close to the camera")
```

Comment:
Here is what I'm talking about: you should actually catch this during field validation. You can use a common validator to `origin` and `target` using the decorator validator syntax.